core/src/avm1/globals/color_transform.rs

```diff
@@ -0,0 +1,299 @@
+use crate::avm1::function::{Executable, FunctionObject};
+use crate::avm1::return_value::ReturnValue;
+use crate::avm1::{Avm1, Object, ScriptObject, TObject, UpdateContext, Value};
+use crate::color_transform::ColorTransform;
+use enumset::EnumSet;
+use gc_arena::MutationContext;
+use std::convert::Into;
+
+pub fn constructor<'gc>(
+    avm: &mut Avm1<'gc>,
+    context: &mut UpdateContext<'_, 'gc, '_>,
+    this: Object<'gc>,
+    args: &[Value<'gc>],
+) -> Result<ReturnValue<'gc>, Error<'gc>> {
+    let red_multiplier = args
+        .get(0)
+        .unwrap_or(&Value::Number(1.into()))
+        .coerce_to_f64(avm, context)?;
+    let green_multiplier = args
+        .get(1)
+        .unwrap_or(&Value::Number(1.into()))
+        .coerce_to_f64(avm, context)?;
+    let blue_multiplier = args
+        .get(2)
+        .unwrap_or(&Value::Number(1.into()))
+        .coerce_to_f64(avm, context)?;
+    let alpha_multiplier = args
+        .get(3)
+        .unwrap_or(&Value::Number(1.into()))
+        .coerce_to_f64(avm, context)?;
+    let red_offset = args
+        .get(4)
+        .unwrap_or(&Value::Number(0.into()))
+        .coerce_to_f64(avm, context)?;
+    let green_offset = args
+        .get(5)
+        .unwrap_or(&Value::Number(0.into()))
+        .coerce_to_f64(avm, context)?;
+    let blue_offset = args
+        .get(6)
+        .unwrap_or(&Value::Number(0.into()))
+        .coerce_to_f64(avm, context)?;
+    let alpha_offset = args
+        .get(7)
+        .unwrap_or(&Value::Number(0.into()))
+        .coerce_to_f64(avm, context)?;
+
+    let ct = ColorTransform {
+        r_mult: red_multiplier as f32,
+        g_mult: green_multiplier as f32,
+        b_mult: blue_multiplier as f32,
+        a_mult: alpha_multiplier as f32,
+        r_add: red_offset as f32,
+        g_add: green_offset as f32,
+        b_add: blue_offset as f32,
+        a_add: alpha_offset as f32,
+    };
+
+    apply_color_transform_to_object(ct, this, avm, context)?;
+
+    Ok(Value::Undefined.into())
+}
+
+pub fn apply_color_transform_to_object<'gc>(
+    color_transform: ColorTransform,
+    object: Object<'gc>,
+    avm: &mut Avm1<'gc>,
+    context: &mut UpdateContext<'_, 'gc, '_>,
+) -> Result<(), Error<'gc>> {
+    object.set("redMultiplier", color_transform.r_mult.into(), avm, context)?;
+    object.set(
+        "greenMultiplier",
+        color_transform.g_mult.into(),
+        avm,
+        context,
+    )?;
+    object.set(
+        "blueMultiplier",
+        color_transform.b_mult.into(),
+        avm,
+        context,
+    )?;
+    object.set(
+        "alphaMultiplier",
+        color_transform.a_mult.into(),
+        avm,
+        context,
+    )?;
+    object.set("redOffset", color_transform.r_add.into(), avm, context)?;
+    object.set("greenOffset", color_transform.g_add.into(), avm, context)?;
+    object.set("blueOffset", color_transform.b_add.into(), avm, context)?;
+    object.set("alphaOffset", color_transform.a_add.into(), avm, context)?;
```

Comment:
Unfortunately we shouldn't round-trip between the engine `ColorTransform` object and the object properties, because this will lose precision from the f64 to f32 (and this will get worse once #193 is fixed). The values should just be stored as properties on its own object native object type, and then we can convert to `color_transform::ColorTransform` only when it is needed.

Try this:
```
var ct = new flash.geom.ColorTransform();
ct.redMultiplier = 8589934592;
trace(ct);
```
Expected: `redMultiplier=8589934592` This PR: `redMultiplier=8589935000`.

This does mean we have to duplicate all of the `concat` logic between engine and this code, but that unfortunately seems to be necessary. (Matrix has the same issue).